chainerx_cc/chainerx/cuda/cuda_device/batch_norm.cc

```diff
@@ -100,79 +103,90 @@ class CudnnBNTensorDescriptor {
     cudnnTensorDescriptor_t desc_{};
 };
 
-class CudaBatchNormForwardBackward : public chainerx::GenericBatchNormForwardBackward {
+struct CudaBatchNormState {
+    CudaBatchNormState(Array x_cont, Array x_mean, Array x_inv_std)
+        : x_cont{std::move(x_cont)}, x_mean{std::move(x_mean)}, x_inv_std{std::move(x_inv_std)} {}
+
+    Array x_cont;
+    Array x_mean;
+    Array x_inv_std;
+};
+
+}  // namespace
+
+class CudaBatchNormForwardOp : public BatchNormForwardOp {
```

Comment:
I was taking inspiration from the cuDNN library since these Ops are quite close to the lower cuDNN level operations. I had another idea about BatchNorm and BatchNormGrad, as in Chainer, but that's a different level of abstraction (including graphs).

Reply:
Will all such ops (including `Dot`, `Exp`, `AMax`, ...) have `ForwardOp` suffix?
If not, what's the criteria to differentiate them?

Reply:
Fixed accordingly with "Grad" suffix for backward ops.